src/Gui/ExpressionCompleter.cpp

```diff
@@ -284,8 +284,9 @@ class ExpressionCompleterModel: public QAbstractItemModel {
         App::Property* prop = nullptr;
         // check if the document is uniquely identified: either the correct index in info.doc
         // OR if, the node is a descendant of the root, its row lands within 0...docsize
-        if(idx>=0 && idx<docSize)
+        if(idx>=0 && idx<docSize) {
             doc = docs[idx/2];
+        }
```

Comment:
Just for curiosity, why are curly braces necessary for single-line ifs? Up to now i did not add them because I saw no need and also because most of FC's code don't have them.

Reply:
Forums discussion of clang-format: https://forum.freecadweb.org/viewtopic.php?f=10&t=72183&p=629874

The gist of it is that it's widely considered a best practice due to some very famous bugs that would have been prevented had this formatting strategy been used.

Reply:
Thanks for the pointer. In this discussion I miss the info what can be harmful of omitting the braces. I don't mean when there are comments and then a single statement, but just a single statement alone, so
```
if (blah)
 do something;
else if (blub)
 do another thing;
...
```

Especially when having "else if" this way improves in my opinion the readability, especially on smaller screen since it saves lines.

Besides this, I recently reverted the automatic addition of braces, not because I am opposed of requiring the braces, but it was a nightmare to work with the VC IDE with this. At first, the addition seemed convenient but on longer coding sessions VC is not able to do this right when copy/pasting code around. Often it adds a closing brace despite it is already there. This cost a lot of time since the compilation error messages don't help. So one has to carefully re-read to find additions braces. On the other hand sometimes it does not add the closing brace. Still a mystery for me. What about you, are you on VC 2022 as well? When you enabled automatic brace insertion, and code around for a while, do you encounter problems? I had a look today in forums but cannot find a way to use a clang file and only overwrite one of its setting.
I see no workaround because I cannot turn of the formatter because then the code would not be formatted at all. And having a custom clang file only for me is error-prone (different PCs, keeping it up to date.)

Reply:
from what you are saying it sounds like it's messing up the file _while_ you are coding, that does sound annoying and error prone, can't you run clang format just before committing?

Reply:
> Thanks for the pointer. In this discussion I miss the info what can be harmful of omitting the braces

my first guess is that later someone might forget to put braces when trying to add more lines inside the if statement, I've certainly done so myself 🤦‍♂️

Reply:
In my opinion, format-as-you-type is more trouble than it is worth, in _any_ IDE. My strategy is to literally not worry about the format at all while I am coding, and just make the code look how I like it to look. Then, before committing, I select and auto-format (or in the case of Python, use Black as a pre-commit hook).

Reply:
"In this discussion I miss the info what can be harmful of omitting the braces."

The most famous case is the "Apple goto fail".   There is also the python mantra "explicit is better than implicit".   From a maintainability standpoint, anything that reduces the chances of misinterpreting code is a good thing.